feat:  add Dev Portal one pager for developers for (#7450)
Kong/docs.konghq.com: app/konnect/dev-portal/register-and-create-app.md

---
title: Register and create an application as a developer
---

> This guide explains how developers can get started with the developer platform by registering and creating an application. 

Reviewer: ```suggestion
Dev Portal
```

> This guide explains how developers can get started with the developer platform by registering and creating an application. 

Reviewer: ```suggestion
This guide explains how developers can get started with the Dev Portal by registering and creating an application. 
```

> The Dev Portal enables developers to integrate applications with APIs from third-party providers. For instance, if you are developing for a company that requires access to two distinct APIs from a third-party provider, you would need to set up an application within the Dev Portal that consumes both APIs.

Reviewer: ```suggestion
The Dev Portal enables developers to quickly get access to their APIs of interest, in a self serve fashion. Whether you're accessing the Dev Portal of a company you work for, or are interested in, the Dev Portal lets you get started, get your credentials, and start accessing those APIs in minutes.
```

The following diagram shows how you can register your app in Dev Portal:

{% mermaid %}
flowchart TD
    A[Sign up for Dev Portal] --> B(Create an application)

>     B --> C(Add products to the application)

Reviewer: ```suggestion
    B --> C(Add APIs to the application)
```

    C --> D(Create application credentials)

>     D --> |Generate in Dev Portal| E[Use application] 

Reviewer: From the developers perspective do they really need to worry about D? B (creating an application) will be informed by their Dev Portal admin--they'll tell them hey use key auth or OIDC or DCR and follow these steps. After that, they don't need to worry about application auth/creation

Author: @c3pko Do developers still need to create credentials if the application uses OIDC or DCR? I think I still might be confused about credentials and auth for developers.

Reviewer: I think this comment was in reference to needing to select or create an auth method? I don't see anything about auth methods so we can disregard my comment maybe I misunderstood the original text

    D --> |Create in IdP| E
{% endmermaid %}

> > **Figure 1:** To create an application against APIs in the Dev Portal, you must first register an account. You can use APIs in the developer platform to create your own app and register that app with the company's developer platform. Before you can use your app, you must create credentials to access the app.

Reviewer: ```suggestion
> **Figure 1:** In order to use the Dev Portal, you must first register for the Portal with an account. You will receive an email from the Portal admin prompting you to confirm your email address.
```

<details><summary>Key concepts</summary>

{% capture konnect_concepts %}

> **Application:** An application enables developers to register and use APIs, whether they are your company’s own or those of a third-party provider. To facilitate diverse integrations, developers can create applications that consume multiple APIs from different providers, streamlining various services into a single offering.

Reviewer: ```suggestion
**Application:** An application enables developers to register for APIs in the Dev Portal. For ease of use, developers can create applications that consume multiple APIs. After creating the application, developers can create a single credential that consumes all the APIs that application is registered for. If you're unable to register more than one API to an application, be sure to check that the authentication strategies of those APIs are the same.
```

**Application credentials:** Credentials authorize developers to access an application. You can generate credentials automatically, or  manage them manually through an Identity Provider (IdP), like OIDC.

> **Developer platform:** The developer platform allows developers like you to locate, access, and consume APIs. Using the Dev Portal, you can browse and search API documentation, test API endpoints, and manage your own credentials. 

Reviewer: ```suggestion
**Developer Portal:** The Dev Portal allows developers like you to locate, access, and consume APIs. Using the Dev Portal, you can browse and search API documentation, test API endpoints, manage your own credentials, and view your API consumption in the Analytics page.
```

{% endcapture %}

{{ konnect_concepts | markdownify }}

</details>

## Register and create an application

All developers must register through the {{site.konnect_short_name}} Dev Portal. A {{site.konnect_short_name}} admin can provide you with the correct registration URL.

1. **Register or sign in to the developer platform**
    
    Navigate to Dev Portal and do one of the following:
    * Register for access by navigating to the Dev Portal and creating an account.
    * Sign in to the Dev Portal with single sign-on (SSO).

>     Once you're approved for access, you can create an application in Dev Portal.

Reviewer: realizing we already have this https://docs.konghq.com/konnect/dev-portal/applications/dev-apps/ so now I'm a bit confused about this one pager. Can we end the doc here and direct people here: https://docs.konghq.com/konnect/dev-portal/applications/dev-apps/. Because there are 2 ways you can create an app in Portal

> 1. **Create an application**

Reviewer: realizing we already have this https://docs.konghq.com/konnect/dev-portal/applications/dev-apps/ so now I'm a bit confused about this one pager. Can we end the doc here and direct people here: https://docs.konghq.com/konnect/dev-portal/applications/dev-apps/. Because there are 2 ways you can create an app in Portal

Reviewer 2: We're removing that page and this new one-pager is replacing it - you can see that `dev-apps.md` is being deleted in this PR.

I did notice that you can create an app when you're browsing an API though, so maybe that needs to be part of this flow somehow.

Reviewer: Can we preserve what exists today (copy over if we have to)?

Author: @c3pko I copied over any of the other info from that old doc into this one (including Kong auto selecting auth strategy and how to edit/delete an app). Was there something specific in https://docs.konghq.com/konnect/dev-portal/applications/dev-apps/ that wasn't covered here? We've been trying to remove some of these CRUD docs throughout the Konnect docs since the create/update/delete process is usually intuitive in the UX.

Reviewer: thanks, lgtm!

    Click **My Apps**. From the **My Apps** page, click **New App** and fill out the application information. 
    
    {:.note}
    > **Note:** The Reference ID must be unique. If your organization is using the [OIDC](/konnect/dev-portal/applications/enable-app-reg#oidc-flow) flow for application registration, enter the ID of your third-party OAuth2 claim.

> 1. **Add products to your application**

Reviewer: ```suggestion
1. **Add APIs to your application**
```

>     Click **Catalog**, click the product version you want to create an application for, and then click **Register**. In the registration pop-up, select the app you just created and click **Request Access**. Repeat this for as many product versions as you want to add to one application.

Reviewer: ```suggestion
    Click **Catalog**, click the API you want to create an application for, and then click **Register**. In the registration pop-up, select the app you just created and click **Request Access**. Repeat this for as many APIs as you want to add to one application.
```

---
title: Register and create an application as a developer
---

This guide explains how developers can get started with the Dev Portal by registering and creating an application. 

The Dev Portal enables developers to quickly get access to their APIs of interest, in a self serve fashion. Whether you're accessing the Dev Portal of a company you work for, or are interested in, the Dev Portal lets you get started, get your credentials, and start accessing those APIs in minutes.

In order to create credentials, developers must first create an application in which store those credentials.

> The following diagram shows how you can register your app in Dev Portal:

Reviewer 2: ```suggestion
The following diagram shows how to register your app in Dev Portal:
```

{% mermaid %}
flowchart TD
    A[Sign up for Dev Portal] --> B(Create an application)
    B --> C(Add APIs to the application)
    C --> D(Create application credentials)
    D --> |Generate in Dev Portal| E[Use application] 
    D --> |Create in IdP| E
{% endmermaid %}

> > **Figure 1:** In order to use the Dev Portal, you must first register for the Portal with an account. You will receive an email from the Portal admin prompting you to confirm your email address.

Reviewer 2: ```suggestion
> _**Figure 1:** To use the Dev Portal, you must first register for the Portal with an account. You will receive an email from the Portal admin prompting you to confirm your email address. Then, you can log in, find the API you want to use, create an application with credentials, and start using the API with those credentials._
```

<details><summary>Key concepts</summary>

{% capture konnect_concepts %}
**Application:** An application enables developers to register for APIs in the Dev Portal. For ease of use, developers can create applications that consume multiple APIs. After creating the application, developers can create a single credential that consumes all the APIs that application is registered for. If you're unable to register more than one API to an application, be sure to check that the authentication strategies of those APIs are the same.

**Application credentials:** Credentials authorize developers to access an application. You can generate credentials automatically, or  manage them manually through an Identity Provider (IdP), like OIDC.

> **Developer Portal:** The Dev Portal allows developers like you to locate, access, and consume APIs. Using the Dev Portal, you can browse and search API documentation, test API endpoints, manage your own credentials, and view your API consumption in the Analytics page.

Reviewer 2: ```suggestion
**Developer Portal:** Using the Dev Portal, you can browse and search API documentation, test API endpoints, manage your own credentials, and view your API consumption in the Analytics page.
```
It doesn't need both sentences, they're essentially saying the same thing.

> {% endcapture %}

Reviewer 2: There's a lot of switching between "developers" and "you" in this section. Can we please stick to one, at least within the same paragraph, or introduce the "you" with examples? Eg "Developers can make applications. For example, you can go here and here and do the thing."

    * Register for access by navigating to the Dev Portal and creating an account.
    * Sign in to the Dev Portal with single sign-on (SSO).

    Once you're approved for access, you can create an application in Dev Portal.

1. **Create an application**

    Click **My Apps**. From the **My Apps** page, click **New App** and fill out the application information. 
    
    {:.note}
    > **Note:** The Reference ID must be unique. If your organization is using the [OIDC](/konnect/dev-portal/applications/enable-app-reg#oidc-flow) flow for application registration, enter the ID of your third-party OAuth2 claim.

1. **Add products to your application**

    Click **Catalog**, click the product version you want to create an application for, and then click **Register**. In the registration pop-up, select the app you just created and click **Request Access**. Repeat this for as many product versions as you want to add to one application.

    Each application can only work with one auth strategy. Dev Portal can contain products with different auth strategies, so keep this in mind when adding more than one product to your application.
    
    {:.note}

>     > **Note:** The Reference ID must be unique. If your organization is using the [OIDC](/konnect/dev-portal/applications/enable-app-reg#oidc-flow) flow for application registration, enter the ID of your third-party OAuth2 claim.

Reviewer 3: I don't think this note needs to be in there twice.

1. **Create API credentials for your application**

    You can generate credentials using one of the following methods:

>         1. To generate an API key from the Dev Portal, navigate to the **My Apps** page and click the application that you want to generate a credential for. 

Reviewer 3: Formatting here is broken
![image](https://github.com/Kong/docs.konghq.com/assets/23319190/4b64cf37-0375-4772-8049-731c9894686a)

---
title: Register and create an application as a developer
---

This guide explains how developers can get started with the developer platform by registering and creating an application. 

> Dev Portal allows developers to create applications by using APIs from another company. For example, say you're a developer for a company that wants to consume two different APIs from another company. To make this possible, you'd need to create an application in Dev Portal that consumes both the APIs.

Reviewer 3: ```suggestion
The Dev Portal enables developers to integrate applications with APIs from third-party providers. For instance, if you are developing for a company that requires access to two distinct APIs from a third-party provider, you would need to set up an application within the Dev Portal that consumes with both APIs.
```

> Dev Portal allows developers to create applications by using APIs from another company. For example, say you're a developer for an airline company. Your airline is forming a partnership with a hotel and a theme park so that you can get a vacation package that includes airplane, theme park, and hotel tickets. To make this possible, you'd need to create an application that consumes both the hotel's API as well as the theme park API so that when customers buy airplane tickets, your application would use the other companies' APIs to also purchase the theme park tickets and hotel.

Author: ```suggestion
Dev Portal allows developers to create applications by using APIs from another company. For example, say you're a developer for a company that wants to consume two different APIs from another company. To make this possible, you'd need to create an application in Dev Portal that consumes both the APIs.
```

> The following diagram shows how you can register your app in Dev Portal:

Reviewer 3: I know this is marked as draft so feel free to ignore this comment until it's review time.

The airline theme here makes this paragraph confusing for me without explicitly adding anything that helps me understand the concept better. If it makes sense, I would make this much more neutral, like: 

You are a developer tasked with creating an application that consumes two separate APIs.

> {% mermaid %}
> flowchart TD
>     A[Sign up for Dev Portal] -->|Access approved| B(Create an application)
>     B --> |Registration approved| C(Add products to the application)
>     C --> D(Create application credentials)
>     D --> |Generate in Dev Portal| E[Use application] 
>     D --> |Create in IdP| E
> {% endmermaid %}

Reviewer 3: ```suggestion
{% mermaid %}
flowchart TD
    A[Sign up for Dev Portal] --> B(Create an application)
    B --> C(Add products to the application)
    C --> D(Create application credentials)
    D --> |Generate in Dev Portal| E[Use application] 
    D --> |Create in IdP| E
{% endmermaid %}
```

Because the approval steps are outside of the control of the dev reading the doc, I think they can be removed.

> **Figure 1:** To create an application against APIs in the Dev Portal, you must first register an account. You can use APIs in the developer platform to create your own app and register that app with the company's developer platform. Before you can use your app, you must create credentials to access the app.

<details><summary>Key concepts</summary>

{% capture konnect_concepts %}

> **Application:** An application allows developers to register with APIs. These can be your company's APIs or the APIs of another company. For example, I'm a developer for an airline company. My airline is forming a partnership with a hotel and a theme park so that you can get a vacation package that includes airplane tickets, theme park tickets, and hotel tickets. To make this possible, I'd need to create an application that consumes both the hotel's API as well as the theme park API so that when they buy airplane tickets, my application would use the other companies' APIs to also purchase the theme park tickets and hotel.

Reviewer 3: ```suggestion
**Application:** An application enables developers to register and use APIs, whether they are your company’s own or those of a third-party provider. To facilitate diverse integrations, developers can create applications that consume multiple APIs from different providers, streamlining various services into a single offering.
```

The airline mention here made it more confusing for me.

> **Application credentials:** Application credentials are what allow the developer to access the app. You can either generate credentials or manually manage them in an IdP with OIDC.

Reviewer 3: ```suggestion
**Application credentials:** Credentials authorize developers to access an application. You can generate credentials automatically, or  manage them manually through an Identity Provider (IdP), like OIDC.
```

Reviewer 3: I said like OIDC here because SAML is coming.

    Click **My Apps**. From the **My Apps** page, click **New App** and fill out the application information. 
    
    {:.note}
    > **Note:** The Reference ID must be unique. If your organization is using the [OIDC](/konnect/dev-portal/applications/enable-app-reg#oidc-flow) flow for application registration, enter the ID of your third-party OAuth2 claim.

1. **Add products to your application**

    Click **Catalog**, click the product version you want to create an application for, and then click **Register**. In the registration pop-up, select the app you just created and click **Request Access**. Repeat this for as many product versions as you want to add to one application.

    Each application can only work with one auth strategy. Dev Portal can contain products with different auth strategies, so keep this in mind when adding more than one product to your application.
    
    {:.note}
    > **Note:** The Reference ID must be unique. If your organization is using the [OIDC](/konnect/dev-portal/applications/enable-app-reg#oidc-flow) flow for application registration, enter the ID of your third-party OAuth2 claim.

1. **Create API credentials for your application**

>     Every application must have credentials so that you can access your app. An application can have multiple credentials. You can generate credentials using one of the following methods:

Reviewer 3: ```suggestion
    You can generate credentials using one of the following methods:
```

>     * **Generate credentials**: A credential, or API key, generated in the {{site.konnect_short_name}} Dev Portal is a 32-character string associated with an application. 
>         1. To generate an API key from the Dev Portal, navigate to the **My Apps** page and click the application that you want to generate a credential for. 
> 
>         1. In the **Authentication** pane, click **Generate Credential**.
>         
>     * **Manually create OIDC credentials**: To use ODIC without DCR, you must manually create the application in an IdP and use the reference ID (listed on your application in Dev Portal) when creating the application.
> 

Reviewer 3: ```suggestion
        1. To generate an API key from the Dev Portal, navigate to the **My Apps** page and click the application that you want to generate a credential for. 

        1. In the **Authentication** pane, click **Generate Credential**.
        
    * **Manually create OIDC credentials**: To use ODIC without DCR, you must manually create the application in an IdP and use the reference ID (listed on your application in Dev Portal) when creating the application.

```

Would it make sense to link to https://deploy-preview-7450--kongdocs.netlify.app/konnect/dev-portal/applications/dynamic-client-registration/

Or any of the OIDC docs for the OIDC step?

    You can use your app once you've added products to you application, created credentials for your application, and a Dev Portal admin has approved your app.

> 1. **Use your application**

Reviewer 3: This doesn't feel like a step to me. I wonder if you could end on the last step and then conclude with a sentence like, 'now you can start using your application'.

>     <!--how do I USE the application?-->

Author: ```suggestion
    You can start making requests to the endpoints in your application, making sure to use the app credentials you configured.
```
